File: common/src/main/scala/com/taskadapter/connector/common/ValueTypeResolver.scala
package com.taskadapter.connector.common

object ValueTypeResolver {

  def getValueAsFloat(value: Any): Float = {
    if (value.isInstanceOf[String]) {
      value.asInstanceOf[String].toFloat
    } else if (value.isInstanceOf[Int]) {
      value.asInstanceOf[Int].toFloat
    } else {
      value.asInstanceOf[Float]
    }
  }

  def getValueAsInt(value: Any): Int = {
    if (value.isInstanceOf[String]) {
      value.asInstanceOf[String].toInt
    } else if (value.isInstanceOf[Float]) {
      value.asInstanceOf[Float].toInt
    } else {
      value.asInstanceOf[Int]
    }
  }

  def getValueAsString(value: Any): String = {
    if (value.isInstanceOf[Seq[String]]) {
      val seq = value.asInstanceOf[Seq[String]]
      if (seq.isEmpty) {
        ""
      } else {
        seq.head
      }
    } else {
      value.asInstanceOf[String]
    }
  }

}

File: common/src/main/java/com/taskadapter/connector/common/ValueTypeResolver.java
package com.taskadapter.connector.common;

import scala.collection.Seq;

public class ValueTypeResolver {
    public static Float getValueAsFloat(Object value) {
        if (value == null) {
            return null;
        }
        if (value instanceof String) {
            return Float.parseFloat((String) value);
        }
        if (value instanceof Integer) {
            return ((Integer) value).floatValue();
        }
        return (Float) value;
    }

    public static Integer getValueAsInt(Object value) {
        if (value == null) {
            return null;
        }
        if (value instanceof String) {
            return Integer.parseInt((String) value);
        }
        if (value instanceof Float) {
            return ((Float) value).intValue();
        }
        return (Integer) value;
    }

    public static String getValueAsString(Object value) {
        if (value == null) {
            return "";
        }

        if (value instanceof Seq) {
            Seq seq = (Seq) value;
            if (seq.isEmpty()) {
                return "";
            }
            return seq.head().toString();
        }

        if (value instanceof String) {
            return (String) value;
        }

        throw new RuntimeException("Cannot parse value " + value);
    }
}
